### ISSUE TITLE
Matched modalities: Train visual-state adapters following #75

### ISSUE THREAD

**Issue by Sino-Huang** [Sino-Huang/multimodality_on_planning#113]
> **Completed — unlabelled 128px v5:** Preparation and both-A100 qualification pass; all twelve matched adapters (four retained text, four visual, four multimodal) are complete and independently verified. Every cell used the same ordered 512-record, seed-17, one-epoch, 16-update contract, with diagnostics at updates 5/10 and trained-adapter/base isolation. The automatic chain finished successfully; the technical decision is CONTINUE. Final evaluation has not run. [Training evidence](https://github.com/Sino-Huang/multimodality_on_planning/blob/d82d210/docs/experiments/matched-modalities/v5-training.json) · [Decision](https://github.com/Sino-Huang/multimodality_on_planning/blob/d82d210/docs/experiments/matched-modalities/v5-decision.json).

**Parent:** #38
**Study:** `matched-modalities-v5`
**Protocol:** [plan](https://github.com/Sino-Huang/multimodality_on_planning/blob/main/docs/experiments/matched-modalities/plan.md), `configs/experiments/matched-modalities/study-v5.json` and `membership.json`.

This is prospective follow-up work. Historical experiments and their failed performance gates remain unchanged. No new authorization manifest is required by this successor; the user has explicitly authorized qualification and bounded training execution.

## What to deliver
Successor to #75; preserve its completed experiment and original outcome.
Produce four fresh `visual-state` SFT adapters from the pinned Qwen base for BFS, BFWS and both additive settings. Share the same ordered 512 records per algorithm with the other modalities: seed 17, one epoch, effective batch 32, exactly 16 optimizer updates, fixed AdamW/cosine/LoRA settings. Evaluate only fixed development diagnostics at updates 5 and 10 and final technical probes. No final-test outcomes or historical checkpoint substitution.

## Acceptance criteria
- Consume #92's technically qualified matched data and study settings. Prove every declared training record was presented once in order and final checkpoints completed update 16.
- Retain checkpoint paths, source task/input bindings, progress, package/model versions and resource accounting; independently verify completion and diagnostics.
- All three training tickets share **16,200 seconds total**, including model loads, saves, diagnostics, probes and failures. Use one shared scheduler/ledger, not an independent per-ticket allowance per ticket. Stage preparation must also admit the six-hour final matrix before training.
- At most one model worker per GPU, explicit distinct MASTER_PORT values, 30-second heartbeats, cumulative resume accounting and hard shutdown deadline. Do not stop unrelated processes.
- Never tune or retrain for a higher score. A zero-score trained adapter is valid evidence if technically sound. An incomplete adapter is not a completed training cell and cannot enter a completed matched comparison.
- Tests cover membership/order/batch/update parity, resume and cutoff accounting, no old-checkpoint substitution and false-completion refusal. Record real execution evidence before closing this experiment ticket as complete; a dry-run alone does not close it.

## Command contract to implement in #92
`python -u scripts/run_matched_modalities.py train --study configs/experiments/matched-modalities/study-v5.json --devices 0 1 --master-ports 18775 18776`

The shared command schedules all modalities; this ticket owns its `visual-state` checkpoints. The command is implemented and its bounded execution is authorized. `--dry-run`, `--resume`, read-only verification and per-stage progress are required.

## Blocked by
#92

**Comment by Sino-Huang** [comment]
The newly authorized training/development goal reached its permitted **verified terminal prerequisite-failure branch**. This training ticket remains **OPEN**; no adapter or development result was produced.

Fresh read-only verification again passed all 2,156 selected input records and independently reproduced all 64 Storage rejections (56 retained-task overlaps, eight exact BFS expansion-limit failures). Preparation remains VALID_STOP/model_input_ready=false, and #92 still lacks the final panel/views, full GPU scheduler with cumulative deadline/resume enforcement, and hardware qualification.

The current training dry-run and actual command were exercised. The generic guard returned VALID_STOP before model loading; the retained downstream report classifies training as **ANCESTOR_STOP** and preserves the original command output. This is not a GPU OOM, a spent training budget, an authorization refusal, or a learned-performance failure.

Coverage is explicitly **0/12 training cells**, zero optimizer updates, zero development diagnostic evaluations, no new checkpoint bindings, and no final model calls. All **18,000 seconds** of the shared training/development allocation remain unspent. Old #75/#76/#67 checkpoints were not substituted; no frozen training settings or task-selection rules were changed.

[Terminal report](https://github.com/Sino-Huang/multimodality_on_planning/blob/main/docs/experiments/matched-modalities/training-admission.md) · [machine-readable accounting](https://github.com/Sino-Huang/multimodality_on_planning/blob/main/docs/experiments/matched-modalities/training-admission.json).

Next prerequisite: revise task selection prospectively and complete #92. The failure report completes only the requested reporting alternative, not this training ticket or the matched experiment. Do not proceed as if a trained matched checkpoint exists.

**Comment by Sino-Huang** [comment]
All four visual-state adapters are complete. Commit d82d210 records the actual checkpoint paths, complete ordered membership/settings verification, epoch 1 / step 16, finite diagnostics at steps 5/10, trained-adapter/base isolation, package/code provenance and cumulative spending. The background chain completed preparation, qualification, training, independent verification and decision with exit code 0. Eight new image adapters trained in 5,232.08 seconds; cumulative qualification/training spending is 4,745.35 / 6,744.42 seconds, within 5,400 / 16,200 allocated. The qualified final estimate of 17,155.46 seconds fits the unspent 21,600-second evaluation allocation. No final held-out model evaluation has run. Evidence: docs/experiments/matched-modalities/v5-training.json and v5-decision.json.
